### DISCUSSION TITLE
Subscription Price Variants?

### DISCUSSION THREAD

**Discussion by aanomm** [swellstores/community#312] [Help] [answered]
Hi Swellsters 👋🏾

- I'm finally testing Swell for the first time after watching from afar forever and I'm certainly impressed; congrats on all you've achieved
- I'm using Horizon theme for testing but might use a custom storefront irl
- For either case (Horizon or custom) is it possible to assign subscription prices according to variants?
- Example: Ideally we'd like to offer a single subscription product (coffee) with multiple variants (Blend, Grind, and Size) and the subscription prices changes accordingly
- You can see this example irl here: https://fucha-subb.vercel.app/products/st-ali

**Comment by ericingram** [comment]
Definitely. Product options automatically generate variants (when chosen). You can pass either options or `variant_id` to a cart/order item, and the price should derived from the variant price.

**Comment by aanomm** [comment]
- how and where do subscription variant prices get set @ericingram?
- in Swell Admin at the individual Product level it appears the Subscription price is not tied to any variant at all?
- i understand how to set variant prices and can see via Horizon theme that different variants trigger different prices
- but i can't see in Swell Admin or on  Horizon's frontend that different variants of the same product can have different subscription prices?

**Reply by ericingram** [reply]
@logeshswell can probably answer

**Reply by aanomm** [reply]
ok, i wait with baited breath (because i'm pitching Swell to a Shopify Plus store soon and need to understand how subscription variants work) . thanks @ericingram

**Reply by logeshswell** [reply]
@fucha-co Can you give me some more time to look into your queries? 
I will keep you posted once I have more details.

**Reply by logeshswell** [reply]
> how and where do subscription variant prices get set?

Variant can accept only standard pricing with additional pricing rules. Subscription pricing can be set only for the base product and not for the variants.

**Reply by logeshswell** [reply]
> in Swell Admin, at the individual Product level it appears the Subscription price is not tied to any variant at all?

Yes, you're right!

**Reply by logeshswell** [reply]
> I can't see in Swell Admin or on Horizon's frontend that different variants of the same product can have different subscription prices?

Yes, it is not feasible as subscription pricing is applicable only for the base product and not the variants.

**Reply by logeshswell** [reply]
@fucha-co If you can share the use case to support this feature, I am happy to take it as a feature request to offer it in the near future.

**Reply by aanomm** [reply]
- hi @logeshswell, and thanks for your support
- @ericingram replied earlier that "yes this is definitely possible" (though perhaps my question was misunderstood?)
- yes the use case is really simply and i have a [live example here](https://fucha-subb.vercel.app/products/st-ali) with Swell's Horizon theme 
- the product is coffee, which is available in different weights/sizes so obviously need the subscription price to change according to the size variant selected

**Reply by ericingram** [reply]
@fucha-co I'm sorry but I misspoke. We actually haven't implemented per-variant subscription pricing in the UI yet. It's a dashboard feature gap, although technically possible via API I believe, but that's probably not what you need.

That said, option value pricing does work. You could set a base price and then an additional price for specific options, which will increase the default variant price when that option it selected.

**Reply by ericingram** [reply]
Here's an example of what it looks like in the dashboard

<img width="1254" alt="image" src="https://github.com/swellstores/community/assets/952134/b7efe31a-e90d-4a34-ac7c-b3f68426cb23">

**Reply by aanomm** [reply]
- ok that's excellent to know, thank you (@ericingram)
- d'you think variant subscription pricing will be a Swell Dashboard option soon, later, or much later?

**Reply by aanomm** [reply]
- the additional price info showing with the Size Option is a nice touch, but not having it atm isn't a deal-breaker provided the frontend can show the appropriate Sub price when a User changes the "Size" variant 
- and given this is available via the api that makes it mostly a matter of coding this into a theme right? i have very competent coder who will be doing the coding, so i just need to know it's possible before we start

**Reply by ericingram** [reply]
Our team is working on a design proposal for it now, but using the option value approach it can definitely be done in the UI as well

**Comment by holden** [comment]
> @fucha-co I'm sorry but I misspoke. We actually haven't implemented per-variant subscription pricing in the UI yet. It's a dashboard feature gap, although technically possible via API I believe, but that's probably not what you need.
> 
> That said, option value pricing does work. You could set a base price and then an additional price for specific options, which will increase the default variant price when that option it selected.

I'm also interested in this feature. How could this be achieved through the API? Even just a few options or variants using subscriptions would be very helpful.

**Reply by holden** [reply]
Or more specifically, how did you accomplish this: https://swell-horizon-demo.vercel.app/products/moisturizer-spf-15

**Reply by ericingram** [reply]
@holden that page is using the option value price that I mentioned above. In the dashboard, click to edit options and then click the option tag itself

**Reply by holden** [reply]
![CleanShot 2023-07-22 at 19 44 38](https://github.com/swellstores/community/assets/48241/ac8f1db4-9675-43b2-b76b-cd03546c4d5a)

Yes, I have this, and it works fine for standard pricing, but just not for subscriptions: https://book.cirqus.co/products/mindfulness/ Am I missing something?

**Reply by aanomm** [reply]
- i'm not sure either @holden 
- i've set up a new demo store to try and understand this more, but i'm stuck on the same subscription variant price issue
- my demo store is a simple use case with minimal variants, a very common subscription product (coffee!), and i've turned standard pricing off to further simplify things; but still i can't set set subscription pricing variants to suite this simple use case
- in my [demo coffee store](https://democoffee.vercel.app/products/coffee) there's just 2 blends of coffee (Standard and Premium) and 3 sizes (Small, Medium, and XLarge)
- obviously the Premium blend coffee is more expensive than the Standard blend so i need i can assign an `Add Price` to the Premium `Option Value`. but the problem is that an `Add Price` only creates a single fixed Add Price (for example $2) which then applies to all 3 size variants. whereas I need a $2 Add Price for SMALL x PREMIUM, a $4 Add Price for MEDIUM x PREMIUM, and a $10 Add Price for XLARGE x PREMIUM

The prices I need to set are...

SMALL x STANDARD = $20
SMALL x PREMIUM = +$2 ($22)

MEDIUM  x STANDARD = $30
MEDIUM x PREMIUM = +$3 ($33)

XLARGE x STANDARD  = $40
XLARGE x PREMIUM = +$4 ($44)

... but as you'll see on my demo store the PREMIUM variations are always fixed at `[STANDARD PRICE] +$2` because only 1 `Add Price` can be assigned — unless i'm just totally lost and wrong?

hopefully @ericingram or @logeshswell can confirm if this is possible asap as i think we're both faced with the same problem

**Reply by ericingram** [reply]
@fucha-co @holden we'll look into this more

**Reply by logeshswell** [reply, marked as answer]
I can confirm that variant subscription pricing is supported by Swell via **Add price** feature in the product options. It is available in the Horizon, but not in the origin theme due to an implementation gap.

> but the problem is that an Add Price only creates a single fixed Add Price

@fucha-co Secondly, Add Price value is a constant value applied for all other options. It is not supported to set multiple added-price values for the same option of different criteria at the moment.

cc: @ericingram

**Reply by logeshswell** [reply]
@holden Implementing the frontend changes in origin theme shouldn't be a big deal as you can refer to the horizon theme code changes for reference.
PRs are welcome to support this feature in the origin theme.
Thanks!

**Reply by aanomm** [reply]
- i appreciate the support and follow-up @logeshswell and @ericingram 
- i understand the point that `add price value` has a specific job and it's indeed doing exactly what it was designed to do
- still it's surprising and disappointing that such a simple use case can't be achieved in Swell

**Reply by ericingram** [reply]
😅
